cmd/metrics.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	httpRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "minio_http_requests_total",
+			Help: "Total number of requests served by current Minio server instance",
+		},
+		[]string{"request_type"},
+	)
+	httpRequestsDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "minio_http_requests_duration_seconds",
+			Help:    "Time taken by requests served by current Minio server instance",
+			Buckets: []float64{.001, .003, .005, .1, .5, 1},
+		},
+		[]string{"request_type"},
+	)
+	networkSentBytes = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "minio_network_sent_bytes",
+			Help: "Total number of bytes sent by current Minio server instance",
+		},
+	)
+	networkReceivedBytes = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "minio_network_received_bytes",
+			Help: "Total number of bytes received by current Minio server instance",
+		},
+	)
+	onlineMinioDisks = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "minio_online_disks_total",
```

Comment:
I would recommend having a metric that represents how many configured disks there are, and then either a gauge of online or offline.  Something like this:

```
minio_total_disks
minio_offline_disks
```

This allows for easier math functions like `minio_offline_disks / minio_total_disks` or `1 - (minio_offline_disks / minio_total_disks)`.  These can also be aggregated for whole clusters with `sum()` functions.